storage/mongo/utils.js

```diff
@@ -0,0 +1,20 @@
+var MongoClient = require('mongodb').MongoClient;
+var mydb;
+
+module.exports = function (config) {
+
+  connectToServer = new Promise(function (resolve, reject) {
```

Comment:
- Use connect to Server as <pre>connectToServer: function(){
return when.promise(function(reject,resolve){     }) }
</pre>

- Here you did not use when.js but `new Promise` which is inbuilt in Nodejs we use `when.js` for it wrap up around new Promise and extended functionalities that it give us. so Please use like.